src/rebar_plugins.erl

```diff
@@ -11,26 +14,35 @@
 %% Public API
 %% ===================================================================
 
--spec install(rebar_state:t()) -> rebar_state:t().
-install(State) ->
-    Plugins = rebar_state:get(State, plugins, []),
-
+-spec project_apps_install(rebar_state:t()) -> rebar_state:t().
+project_apps_install(State) ->
+    Profiles = rebar_state:current_profiles(State),
     ProjectApps = rebar_state:project_apps(State),
 
-    OtherPlugins = lists:flatmap(fun(App) ->
-                                         AppDir = rebar_app_info:dir(App),
-                                         C = rebar_config:consult(AppDir),
-                                         S = rebar_state:new(rebar_state:new(), C, AppDir),
-                                         rebar_state:get(S, plugins, [])
-                                 end, ProjectApps),
-
-    handle_plugins(Plugins++OtherPlugins, State).
+    lists:foldl(fun(Profile, StateAcc) ->
+                        Plugins = rebar_state:get(State, {plugins, Profile}, []),
+                        handle_plugins(Profile, Plugins, StateAcc),
+                        lists:foldl(fun(App, StateAcc1) ->
+                                              AppDir = rebar_app_info:dir(App),
+                                              C = rebar_config:consult(AppDir),
+                                              S = rebar_state:new(rebar_state:new(), C, AppDir),
+                                              Plugins2 = rebar_state:get(S, {plugins, Profile}, []),
+                                              handle_plugins(Profile, Plugins2, StateAcc1)
+                                      end, StateAcc, ProjectApps)
+                end, State, Profiles).
```

Comment:
This should be split up into a bunch of functions, I think, but I can take a look later if you just want to get this out.